lib/publiccloud/instance.pm

```diff
@@ -349,6 +349,41 @@ sub wait_for_guestregister {
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 update_instance_ip
+
+    update_instance_ip(timeout => 600)
+
+This subroutine checks the public IP cloud provider provides for the VM.
+When the IP differs from `$self->public_ip` we update `$self->public_ip`.
+
+=cut
+
+sub update_instance_ip {
+    my ($self, %args) = @_;
+    $args{timeout} //= 600;
+    my $delay = $args{timeout} > 180 ? 5 : 1;
+
+    # skip SLES4SAP as incompatible with get_public_ip
+    if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        my $public_ip_from_provider = $self->provider->get_public_ip();
+        
+        my $duration;
+        my $start_time = time();
+        until ($public_ip_from_provider !~ /null/ || ($duration = time() - $start_time) >= $args{timeout}) {
```

Comment:
We don't need here some variable needed in the other case, like even `%args`, only $self in input, and I'd dare to make timeout constant:

```suggestion
    my $self = shift;
    my $timeout = 300;
    my $delay = 5;
    my $public_ip_from_provider = $self->provider->get_public_ip();

    return if (get_var('PUBLIC_CLOUD_SLES4SAP');) {
    until ($public_ip_from_provider !~ /null/ || (time() - $start_time) >= 300 {
    ...
```

Reply:
Sorry, didn't note 😄 Thanks